docs/example-apps/rails.md

```diff
@@ -71,8 +56,42 @@ And you can deploy your application to production
 convox deploy
 ```
 
+You need to define the secret key for your production application
+
+```
+convox env set SECRET_KEY_BASE="$(rails secret)"
+```
+
 Finally you can retrieve the URL from your production application with
 
 ```bash
 convox services
 ```
+
+### Custom Application Components
+
+#### [Dockerfile](https://github.com/convox-examples/rails/blob/master/Dockerfile)
+
+Starting from the [ruby:3.0.0](https://hub.docker.com/_/ruby/) image, the `Dockerfile` defines the steps necessary to turn the application code into an image that is ready to run. 
+
+This `Dockerfile` has 3 steps and they are executed in a particular order to take advantage of Docker's build caching behavior.
+
+1. `bundle install` and `yarn install` are run to install dependencies after copying just the files needed to run these commands. This will ensure that the output these commands are cached unless one of these files changes.
+
+2. The application source is copied over. These files will change frequently so this step of the build will very rarely be cached.
+
+3. Finally, after setting the appropriate environment variables the assets are precompiled.
```

Comment:
> Finally, after setting the appropriate environment variables the assets are precompiled.

I'd move the comma from "Finally," to "variables,".

Reply:
👍🏻